test/BPINN_Tests.jl

```diff
@@ -0,0 +1,221 @@
+
+# ALL SAMPLES PLOTS--------------------------------------
+# Plot problem and its solution
+# using numerical derivatives
+plot(title = "Problem1 y'(x,t),y(x,t) for ODE,BPINN", legend = :outerbottomright)
+physsol1 = [linear_analytic(prob.u0, p, t[i]) for i in eachindex(t)]
+physsol2 = [linear(physsol1[i], p, t[i]) for i in eachindex(t)]
+plot!(t, physsol1, label = "y(x,t)")
+plot!(t, physsol2, label = "y'(x,t)")
+
+# Plot each iv'th sample in samples
+function realsol(out, t, iv)
+    outpreds = out(t')
+    physsol3 = prob.u0 .+ (t .- prob.tspan[1]) .* vec(outpreds)
+    plot!(t, physsol3, label = "y(x,t) $iv th curve")
+end
+
+# Plot all NN parameter samples curve posibilities
+for i in eachindex(fhsamples)
+    out = re(fhsamples[i])
+    realsol(out, t, i)
+end
+
+# added this as above plot takes time this lower line renders the above plot faster
+plot!(title = "Problem1 y'(x,t),y(x,t) for ODE,B PINN", legend = :outerbottomright)
+
+# PLOT MCMC Chain
+plot(fh_mcmc_chain)
+
+# using autodiff
+plot(title = "Problem1 y'(x,t),y(x,t) for ODE,BPINN", legend = :outerbottomright)
+physsol1 = [linear_analytic(prob.u0, p, t[i]) for i in eachindex(t)]
+physsol2 = [linear(physsol1[i], p, t[i]) for i in eachindex(t)]
+plot!(t, physsol1, label = "y(x,t)")
+plot!(t, physsol2, label = "y'(x,t)")
+
+for i in eachindex(fhsamples1)
+    out = re(fhsamples1[i])
+    realsol(out, t, i)
+end
+
+# added this as above plot takes time this lower line renders the above plot faster
+plot!(title = "Problem1 y'(x,t),y(x,t) for ODE,B PINN", legend = :outerbottomright)
+
+# PLOT MCMC Chain
+plot(fh_mcmc_chain1)
+
+# parameter estimation(inverse problems)
+# prob 1
+function lotka_volterra(u, p, t)
+    # Model parameters.
+    α, β, γ, δ = p
+    # Current state.
+    x, y = u
+
+    # Evaluate differential equations.
+    dx = (α - β * y) * x # prey
+    dy = (δ * x - γ) * y # predator
+
+    return [dx, dy]
+end
+
+u0 = [1.0, 1.0]
+p = [1.5, 1.0, 3.0, 1.0]
+tspan = (0.0, 10.0)
+prob = ODEProblem(lotka_volterra, u0, tspan, p)
+solution = solve(prob, Tsit5(); saveat = 0.1)
+# Plot simulation.
+plot(solution)
+time = solution.t
+u = hcat(solution.u...)
+# BPINN AND TRAINING DATASET CREATION, NN create, Reconstruct
+x = u[1, :] + 0.5 * randn(length(u[1, :]))
+y = u[2, :] + 0.5 * randn(length(u[1, :]))
+dataset = [x[1:50], y[1:50], time[1:50]]
+scatter!(time, [x, y])
+
+# prob 2
+linear = (u, p, t) -> -u / 5 + exp(-t + p[1] / 5) * cos(t) / p[2]
+tspan = (0.0, 10.0)
+u0 = 0.0
+p = [2.0, 5.0]
+prob = ODEProblem(linear, u0, tspan, p)
+
+sol1 = solve(prob, Tsit5(); saveat = 0.1)
+u = sol1.u[1:50]
+time = sol1.t[1:50]
+plot(sol1.t, sol1.u)
+
+x̂ = collect(Float64, Array(u) + 0.005 * randn(size(u)))
+dataset = [x̂, time]
+plot!(time, x̂)
+
+# chainfh = Flux.Chain(Dense(1, 8, sigmoid_fast), Dense(8, 2))
+chainfh = Flux.Chain(Dense(1, 5, sigmoid), Dense(5, 1))
+
+fh_mcmc_chain1, fhsamples1, fhstats1 = ahmc_bayesian_pinn_ode(prob, chainfh, dataset,
```

Comment:
Since you are redefining `prob` `dataset` etc you don't expect this one to work for LV? You are passing the `param` for LV here hence the confusion.

Reply:
yeah, corrected that. i still need to clean up the BPINN test file alot